### ISSUE TITLE
Extraneous "expected an assignment or function call" on multiline ||s

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#1169]
### Bug Report 

- __TSLint version__: `3.8.1`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class Test {
    constructor() {
        this.aaa()
        || this.bbb()
        || this.ccc();
    }

    aaa() {
        return false;
    }
    bbb() {
        return false;
    }
    ccc() {
        return false;
    }
}
```

with `tslint.json`: *none (default ruleset)*

#### Actual behavior

**`test.ts[3, 9]: expected an assignment or function call`**

#### Expected behavior

Not that error!

**Comment by JKillian** [comment]
Closing as dupe of #1413. (I know this one is older, but the other issue now has more relevant discussion)
